tensorflow_lite_support/python/task/audio/core/tensor_audio.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2022 The TensorFlow Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""TensorAudio class."""
+
+import numpy as np
+
+from tensorflow_lite_support.python.task.audio.core.pybinds import _pywrap_audio_buffer
+from tensorflow_lite_support.python.task.audio.core import audio_record
+
+_CppAudioFormat = _pywrap_audio_buffer.AudioFormat
+_LoadAudioBufferFromFile = _pywrap_audio_buffer.LoadAudioBufferFromFile
+
+
+class TensorAudio(object):
+  """A wrapper class to store the input audio."""
+
+  def __init__(self,
+               audio_format: _CppAudioFormat,
+               buffer_size: int) -> None:
+    """Initializes the `TensorAudio` object.
+
+    Args:
+      audio_format: format of the audio.
+      buffer_size: buffer size of the audio.
+    """
+    self._format = audio_format
+    self._buffer_size = buffer_size
+    self._buffer = np.zeros(
+        [self._buffer_size, self._format.channels], dtype=np.float32)
+
+  def clear(self):
+    """Clear the internal buffer and fill it with zeros."""
+    self._buffer.fill(0)
+
+  @classmethod
+  def create_from_wav_file(cls,
+                           file_name: str,
+                           buffer_size: int) -> "TensorAudio":
+    """Creates `TensorAudio` object from the WAV file.
+
+    Args:
+      file_name: WAV file name.
+      buffer_size: Required input buffer size.
+
+    Returns:
+      `TensorAudio` object.
+
+    Raises:
+      status.StatusNotOk if the audio file can't be decoded.
+    """
+    # TODO(b/220931229): Raise RuntimeError instead of status.StatusNotOk.
+    # Need to import the module to catch this error:
+    # `from pybind11_abseil import status`
+    # see https://github.com/pybind/pybind11_abseil#abslstatusor.
+    audio = _LoadAudioBufferFromFile(
+      file_name, buffer_size, np.zeros([buffer_size]))
+    tensor = TensorAudio(audio.audio_format, audio.buffer_size)
+    tensor.load_from_array(np.array(audio.float_buffer, copy=False))
+    return tensor
+
+  def load_from_audio_record(self, record: audio_record.AudioRecord) -> None:
```

Comment:
Why can't we use this as a classmethod and name it as create_from_audio_record?

Reply:
The logic here is that:
* TensorAudio buffer size must be the same as the AudioClassifer/AudioEmbedder model's input size. 
* Therefore, TensorAudio instances should be created by the AudioClassifer/AudioEmbedder object. The input size can be transparent to the model user, and they only need to *load* the data into the TensorAudio object without worrying about the input data size.
* AudioRecord object doesn't know anything about the TFLite model. Therefore, creating a TensorAudio from an AudioRecord instance isn't useful because it may not the compatible with the TFLite model that the user wants to use.

This behavior is similar to that of the Java's TensorAudio, but there's only one difference is that the Java's TensorAudio uses a ring buffer, while this Python's TensorAudio only implements a simple buffer.

@kinaryml After re-reading the [Java's TensorAudio implementation](AudioClassifer/AudioEmbedder), I think we should implement the feature of the TensorAudio as a ring buffer. In particular:
* load_from_array should be able to load an arbitrary size array, similar to the `public void load(float[] newData, int offset, int size)` of the Java implementation. 
  * If the size of the array is larger than the TensorAudio buffer size, only load the most recent data from the array.
  * If the size of the array is smaller than the TensorAudio buffer size, roll the internal buffer backward and add the new data to the end of the buffer (similar to the AudioRecord buffer implementation).
* load_from_array should support `offset` and `size` as optional parameters.

@ziyeqinghan @lu-wang-g @mbrenon WDYT?